packages/react-checkbox/src/components/Checkbox/Checkbox.types.ts

```diff
@@ -1,34 +1,111 @@
 import * as React from 'react';
-import { ComponentPropsCompat, ComponentStateCompat } from '@fluentui/react-utilities';
+import { ComponentPropsCompat, ComponentStateCompat, ShorthandProps } from '@fluentui/react-utilities';
+import { LabelProps } from '@fluentui/react-label';
 
 /**
  * Checkbox Props
  */
-export interface CheckboxProps extends ComponentPropsCompat, React.HTMLAttributes<HTMLElement> {
-  /*
-   * TODO Add props and slots here
-   * Any slot property should be listed in the checkboxShorthandProps array below
-   * Any property that has a default value should be listed in CheckboxDefaultedProps as e.g. 'size' | 'icon'
+export interface CheckboxProps
+  extends Omit<ComponentPropsCompat, 'children'>,
+    Omit<React.HTMLAttributes<HTMLElement>, 'defaultChecked'> {
+  /**
+   * Label that will be rendered next to the input.
+   */
+  label?: ShorthandProps<LabelProps>;
+
+  /**
+   * Indicator to be rendered as the checkbox icon.
+   */
+  indicator?: ShorthandProps<ComponentPropsCompat>;
+
+  /**
+   * Hidden input that handles the checkbox's functionality.
+   */
+  input?: ShorthandProps<React.InputHTMLAttributes<HTMLInputElement> & { ref: React.RefObject<HTMLInputElement> }>;
+
+  /**
+   * Disabled state of the checkbox.
+   */
+  disabled?: boolean;
+
+  /**
+   * Required state of the checkbox.
+   */
+  required?: boolean;
+
+  /**
+   * A checkbox can be rendered with a circular shape.
+   */
+  circular?: boolean;
+
+  /**
+   * A checkbox's state can be controlled.
+   * @defaultvalue false
+   */
+  checked?: 'mixed' | boolean;
+
+  /**
+   * Whether the checkbox should be rendered as checked by default.
    */
+  defaultChecked?: 'mixed' | boolean;
+
+  /**
+   * Checkbox supports two different checkbox sizes.
+   * @defaultvalue 'medium'
+   */
+  size?: 'medium' | 'large';
+
+  /**
+   * Determines whether the label should be positioned before (start) or after (end) the checkbox.
+   * @defaultvalue 'end'
+   */
+  labelPosition?: 'start' | 'end';
+
+  /**
+   * ID of the root element that wraps the checkbox and label.
+   */
+  rootId?: string;
+
+  /**
+   * ID of the native element that represents the checkbox.
+   */
+  id?: string;
+
+  /**
+   * Callback to be called when the checked state value changes.
+   */
+  onChange?: (ev?: React.FormEvent<HTMLElement | HTMLInputElement>, data?: CheckboxOnChangeData) => void;
```

Comment:
I might be missing it, but I don't see where you're calling this callback?

Reply:
Before making `input` a slot it was used but now that `input` is a slot there's no need for it. I will remove it.